llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+            write(deleteROSpec, "DELETE_ROSPEC");
+            pause(200);
+
+            // Not sure if needed, but pause before shutting down
+            pause(500);
+
+            // Send a CLOSE_CONNECTION to with the reader
+            CLOSE_CONNECTION cc = new CLOSE_CONNECTION();
+            write(cc, "CLOSE_CONNECTION");
+
+            // Wait to receive the CLOSE_CONNECTION_RESPONSE which closes the thread
+            synchronized (rt) {
+                try {
+                    systemPrint("\nWaiting for the Reader response of closure...");
+                    rt.wait();
+                } catch (InterruptedException e) {
+                    systemPrint("\nProcess interupted, aborting...");
+                }
+            }
+
+            // set reader to Offline
+            readerOffline = true;
+        }
+
+    }
+
+    /**
+     * Notification that the connection to the Vantiq Source is either online.
+     *
+     */
+    public void vantiqSourceConnectionOnline() {
+        vantiqSourceOffline = false;
+    }
+
+    /**
+     * Notification that the connection to the Vantiq Source is either offline.
+     *
+     */
+    public void vantiqSourceConnectionOffline() {
+        vantiqSourceOffline = true;
+    }
+
+    /**
+     * Change the status of the Reader and send a message to Vantiq if logging.
+     *
+     * @param offlineStatus Indicates if reader is offline (true) or online (false)
+     */
+    public void changeReaderStatus(Boolean offlineStatus) {
+
+        readerOffline = offlineStatus;
+
+        // send a formatted data message to Vantiq
+        Map<String, Object> vantiqMessage = new HashMap<>();
+        vantiqMessage.put("eventType", READER_STATUS_EVENT_NAME);
+        vantiqMessage.put("readerId", readerId);
+        vantiqMessage.put("readerOnline", !readerOffline);
+        this.llrpCore.sendMessage(vantiqMessage);
+
+        if (readerOffline)
+            sendLogMessage(INFO, "Reader is OFFLINE");
+        else
+            sendLogMessage(INFO, "Reader is ONLINE");
+
+    }
+
+    /**
+     * Check the response for the LLRPStatus
+     *
+     * @param status    The Response message to interrogate for the LLRPStatus
+     * @return String   Either "Success" or "(StatusCode): (ErrorDescription)"
+     */
+    private String getLLRPStatus(LLRPStatus status) {
+
+        if (status == null) {
+            return "Invalid Status: null";
+        } else if (status.getStatusCode().toString() == M_SUCCESS) {
+            return "Success";
+        } else {
+            return status.getStatusCode().toString()+": " + status.getErrorDescription();
+        }
+    }
+
+    /**
+     * Method used to include the hostname as a prefix to normally a
+     * System.out.println message.  Since there could be multiple sources
+     * connected to the same connector, it helps with readability of the logs.
+     */
+    private void systemPrint(String msg) {
+        if (msg.startsWith("\n"))
```

Comment:
generally, string manipulation can be done by the log statement, I think.


```log.info("{}: {}", hostname, msg)``` does the same thing as the final log statement, and is much cleaner.

Reply:
yes :)